### PR TITLE
darwin.iproute2mac: init at 1.2.1

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
This adds `iproute2mac`, a CLI wrapper for basic network utilites on Mac OS X inspired with iproute2 on Linux.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @LnL7

### CODE REVIEW

File: pkgs/os-specific/darwin/iproute2mac/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  src = fetchFromGitHub {
+    owner = "brona";
+    repo = "iproute2mac";
+    rev = "v${version}";
+    sha256 = "1n6la7blbxza2m79cpnywsavhzsdv4gzdxrkly4dppyidjg6jy1h";
+  };
+
+  buildInputs = [ python ];
+
+  __impureHostDeps = [
```
Comment (matthewbauer) [inline pkgs/os-specific/darwin/iproute2mac/default.nix]: This is only necessary if you're actually calling these in the builder. You should be able to leave this out.

Comment (flokli) [reply pkgs/os-specific/darwin/iproute2mac/default.nix]: ok, wasn't aware. will strip them out.

Comment (LnL7) [reply pkgs/os-specific/darwin/iproute2mac/default.nix]: Yeah, and I'm not even sure if `/usr/bin` is allowed.

Comment (flokli) [reply pkgs/os-specific/darwin/iproute2mac/default.nix]: @LnL7 from the builder? Don't know too, I just thought I should probably document impure usages somewhere. PR updated with `__impureHostDeps` removed.

File: pkgs/os-specific/darwin/iproute2mac/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    owner = "brona";
+    repo = "iproute2mac";
+    rev = "v${version}";
+    sha256 = "1n6la7blbxza2m79cpnywsavhzsdv4gzdxrkly4dppyidjg6jy1h";
+  };
+
+  buildInputs = [ python ];
+
+  postPatch = ''
+    substituteInPlace src/ip.py --replace /usr/bin/python ${python}/bin/python
```
Comment (matthewbauer) [inline pkgs/os-specific/darwin/iproute2mac/default.nix]: Can you also wrap the other commands here? darwin.network_cmds should have them all.

**Review by matthewbauer — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/darwin/iproute2mac/default.nix (hunk 1 above)
Comment (LnL7) [reply pkgs/os-specific/darwin/iproute2mac/default.nix]: For reference, these are the only prefixes that this works for unless this is overridden in nix.conf with `allowed-impure-host-deps`. 
https://github.com/NixOS/nix/blob/e7cb2847ab1cec48eac6a86c56885b3f0df76275/src/libstore/globals.cc#L22

This is used to open up the build sandbox when sandboxing is enabled, allowing builds to open up arbitrary paths would be a security concern.

File: pkgs/os-specific/darwin/iproute2mac/default.nix (hunk 2 above)
Comment (flokli) [reply pkgs/os-specific/darwin/iproute2mac/default.nix]: right. what about sudo?

File: pkgs/os-specific/darwin/iproute2mac/default.nix (hunk 1 above)
Comment (flokli) [reply pkgs/os-specific/darwin/iproute2mac/default.nix]: thanks for the explanation!

File: pkgs/os-specific/darwin/iproute2mac/default.nix (hunk 2 above)
Comment (matthewbauer) [reply pkgs/os-specific/darwin/iproute2mac/default.nix]: It's probably okay to leave for now. I don't think it's packaged anywhere? /cc @LnL7

**Review by matthewbauer — APPROVED** [review]

File: pkgs/os-specific/darwin/iproute2mac/default.nix (hunk 2 above)
Comment (flokli) [reply pkgs/os-specific/darwin/iproute2mac/default.nix]: patched all network_cmds. I doubt `pkgs/tools/security/sudo` would easily be portable for darwin, so relying on `/usr/bin/sudo` being there might be ok

Comment (LnL7) [reply pkgs/os-specific/darwin/iproute2mac/default.nix]: Sudo can't work, it needs a setuid bit.

**Review by LnL7 — APPROVED** [review]

**Comment by flokli** [comment]
Thanks!

**Pull request merged**